tests/integration/test_studio_editable.py

```diff
@@ -0,0 +1,299 @@
+        self.set_scenario_xml('<editable />')
+        self.go_to_view("studio_view")
+        self.fix_js_environment()
+
+    def test_no_changes_with_defaults(self):
+        """
+        If we load the edit form and then save right away, there should be no changes.
+        """
+        block = self.load_root_xblock()
+        orig_values = {field_name: getattr(block, field_name) for field_name in EditableXBlock.editable_fields}
+        self.click_save()
+        for field_name in EditableXBlock.editable_fields:
+            self.assertEqual(getattr(block, field_name), orig_values[field_name])
+            self.assertFalse(block.fields[field_name].is_set_on(block))
+
+    def test_no_changes_with_values_set(self):
+        """
+        If the XBlock already has explicit values set, and we load the edit form and then save
+        right away, there should be no changes.
+        """
+        block = self.load_root_xblock()
+        block.color = "green"
+        block.count = 5
+        block.comment = "Hello"
+        block.save()
+        orig_values = {field_name: getattr(block, field_name) for field_name in EditableXBlock.editable_fields}
+        # Reload the page:
+        self.go_to_view("studio_view")
+        self.fix_js_environment()
+
+        self.click_save()
+        block = self.load_root_xblock()  # Need to reload the block to bypass its cache
+        for field_name in EditableXBlock.editable_fields:
+            self.assertEqual(getattr(block, field_name), orig_values[field_name])
+            self.assertTrue(block.fields[field_name].is_set_on(block))
+
+    def test_explicit_overrides(self):
+        """
+        Test that we can override the defaults with the same value as the default, and that the
+        value will be saved explicitly.
+        """
+        block = self.load_root_xblock()
+        for field_name in EditableXBlock.editable_fields:
+            self.assertFalse(block.fields[field_name].is_set_on(block))
+
+        color_control = self.get_element_for_field('color')
+        color_control.clear()
+        color_control.send_keys('red')
+
+        count_control = self.get_element_for_field('count')
+        count_control.clear()
+        count_control.send_keys('42')
+
+        comment_control = self.get_element_for_field('comment')
+        comment_control.send_keys('forcing a change')
+        comment_control.clear()
+
+        self.click_save()
+        for field_name in EditableXBlock.editable_fields:
+            self.assertEqual(getattr(block, field_name), block.fields[field_name].default)
+            self.assertTrue(block.fields[field_name].is_set_on(block))
+
+    def test_set_and_reset(self):
+        """
+        Test that we can set values, save, then reset to defaults.
+        """
+        block = self.load_root_xblock()
+        for field_name in EditableXBlock.editable_fields:
+            self.assertFalse(block.fields[field_name].is_set_on(block))
```

Comment:
these 3 lines are present in 2 tests so far - maybe they should be part of a helper method?